### PR TITLE
add eventLastSeen nimbus event store query

### PR DESCRIPTION
**Pull request by jeddai**
This adds a new event store query that will be helpful for the notification message surface moving forward.

[EXP-2995](https://mozilla-hub.atlassian.net/browse/EXP-2995)

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[ff-android: firefox-android-branch-name]` and/or `[fenix: fenix-branch-name]` to the PR title.

### CODE REVIEW

**Review by jhugman — APPROVED** [review]
Sorry this took some time to review.

✅ with comments! No need to hold up on another review cycle.

File: components/nimbus/src/behavior.rs
Comment (jeddai) [reply components/nimbus/src/behavior.rs:314]: Actually I think my brain wasn't firing on all cylinders when writing this -- I have access to the original `Self` in the evaluator, because this method is being called on it. I'm going to remove `Self` from the tuple return.

File: components/nimbus/src/behavior.rs
Comment (jeddai) [reply components/nimbus/src/behavior.rs:323]: That's a great suggestion, I'll do that!

**Pull request merged**
